rosbag2_storage_sqlite3/src/rosbag2_storage_sqlite3/sqlite_storage.cpp

```diff
@@ -770,14 +770,27 @@ void SqliteStorage::add_topic_to_metadata(
     // Before db_schema_version_ = 3 we didn't store metadata in the database and real
     // metadata_.version will be lower than 9
     offered_qos_profiles_str, (db_schema_version_ >= 3) ? metadata_.version : 8);
-  metadata_.topics_with_message_count.push_back(
-    {
-      {
-        get_or_generate_extern_topic_id(inner_topic_id), topic_name, topic_type, ser_format,
-        offered_qos_profiles, type_hash
-      },
-      static_cast<size_t>(msg_count)
+  const rosbag2_storage::TopicMetadata topic_metadata {
+    static_cast<uint16_t>(inner_topic_id), topic_name, topic_type, ser_format,
+    offered_qos_profiles, type_hash};
```

Comment:
```suggestion
  const rosbag2_storage::TopicMetadata topic_metadata {
    get_or_generate_extern_topic_id(inner_topic_id), topic_name, topic_type, ser_format,
    offered_qos_profiles, type_hash};
```
Ups found a bug.

Reply:
yeah right, that was my oversight sorry about that. addressed in https://github.com/ros2/rosbag2/pull/1725/commits/1708c92b0b6e57358325aef392c7e1548ecef320